java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -478,14 +478,15 @@ public void reAlloc() {
    * @throws OutOfMemoryException if the internal memory allocation fails
    */
   public void reallocDataBuffer() {
-    long baseSize = lastValueAllocationSizeInBytes;
     final int currentBufferCapacity = valueBuffer.capacity();
-
-    if (baseSize < (long) currentBufferCapacity) {
-      baseSize = (long) currentBufferCapacity;
+    long newAllocationSize = currentBufferCapacity * 2;
```

Comment:
would +1 here make sense since it gets rounded up to next power of 2 anyways below?

Reply:
The +1 is required only for the offsets vector since it needs to hold one additional entry than the data vector. i.e if the data vector has three elements of sizes 3, 5, 4 - the offsets vector has 4 entries [0, 3, 8, 12]

Reply:
Makes sense, for some reason I thought this could potentially resize to larger then desired because it would get doubled again in nextPowerOfTwo